### PR TITLE
Board place

### PR DESCRIPTION
**Pull request by lpile**
What does this PR do?
--
I've added a way to create valid coordinate arrays which can be used to test `valid_coordinate?` method.

I've also split the test_consecutive_coordinates tests into test_horizontal_consec & test_vertical_consec.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -10,9 +10,40 @@ def initialize
          @cells[index] = Cell.new(index)
        end
      end
+
+		 ###########
+		 @all_cells = []
+		 @all_cells = @cells.keys
```
Comment (BrennanAyers) [inline lib/board.rb]: This is redundant, I think we just need the second line, since it's making an array :)

File: lib/board.rb [hunk 2]
```diff
@@ -10,9 +10,40 @@ def initialize
...
+		 @hor_arr = []
```
Comment (BrennanAyers) [inline lib/board.rb]: We also already have this inside of the valid_placement? method, so we don't need it as an instance variable

File: lib/board.rb [hunk 3]
```diff
@@ -10,9 +10,40 @@ def initialize
...
+		elsif ship.length == 2
+			@hor_arr = []
+	    @all_cells.each_cons(2).map do |x,y|
+				if x.ord == y.ord
+					@hor_arr << [x, y]
+				end
+			end
+			@hor_arr.any? {|valid| valid == coordinates}
+		else
+			raise ArgumentError.new("You messed up!")
```
Comment (BrennanAyers) [inline lib/board.rb:42]: Nice! This is helpful for now while we only have two ship sizes.

File: test/board_test.rb [hunk 4]
```diff
@@ -24,4 +25,35 @@ def test_if_coordinate_is_valid_on_board
...
+
+  def test_if_coordinates_are_consecutive_vertical_on_board
+    refute @board.valid_placement?(@cruiser, ["A1", "B1", "D1"])
+    refute @board.valid_placement?(@submarine, ["A1", "C1"])
+    refute @board.valid_placement?(@cruiser, ["D4", "C4", "B4"])
+    refute @board.valid_placement?(@submarine, ["C1", "B1"])
+
+    assert @board.valid_placement?(@cruiser, ["A1", "B1", "C1"])
+    assert @board.valid_placement?(@submarine, ["C1", "D1"])
+  end
```
Comment (BrennanAyers) [inline test/board_test.rb:57]: I like this, I think extra test coverage for something like this is important while we build.

File: lib/board.rb (hunk 1 above)
Comment (BrennanAyers) [inline lib/board.rb:15]: This is redundant, I think we just need the second line, since it's making an array :)

**Review by BrennanAyers — CHANGES_REQUESTED** [review]

File: lib/board.rb (hunk 2 above)
Comment (BrennanAyers) [inline lib/board.rb]: We also already have this inside of the valid_placement? method, so we don't need it as an instance variable

**Review by BrennanAyers — CHANGES_REQUESTED** [review]

File: lib/board.rb (hunk 2 above)
Comment (lpile) [reply lib/board.rb]: Understandable!

File: lib/board.rb (hunk 1 above)
Comment (lpile) [reply lib/board.rb:15]: K, I understand.

**Review by BrennanAyers — APPROVED** [review]

**Pull request merged**